src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -1,48 +1,19 @@
 (ns puppetlabs.services.jruby.jruby-puppet-schemas
-  (:require [schema.core :as schema]
-            [puppetlabs.services.jruby.puppet-environments :as puppet-env])
-  (:import (clojure.lang Atom Agent IFn PersistentArrayMap PersistentHashMap)
-           (com.puppetlabs.puppetserver PuppetProfiler JRubyPuppet EnvironmentRegistry)
-           (com.puppetlabs.puppetserver.pool LockablePool)
-           (org.jruby Main Main$Status RubyInstanceConfig)
+  (:require [schema.core :as schema])
+  (:import (org.jruby Main Main$Status RubyInstanceConfig)
            (com.puppetlabs.puppetserver.jruby ScriptingContainer)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Schemas
 
-(def pool-queue-type
-  "The Java datastructure type used to store JRubyPuppet instances which are
-  free to be borrowed."
-  LockablePool)
-
-(defrecord PoisonPill
-  ;; A sentinel object to put into a pool in case an error occurs while we're trying
-  ;; to populate it.  This can be used by the `borrow` functions to detect error
-  ;; state in a thread-safe manner.
-  [err])
-
-(defrecord RetryPoisonPill
-  ;; A sentinel object to put into an old pool when we swap in a new pool.
-  ;; This can be used to build `borrow` functionality that will detect the
-  ;; case where we're trying to borrow from an old pool, so that we can retry
-  ;; with the new pool.
-  [pool])
-
-(defrecord ShutdownPoisonPill
-  ;; A sentinel object to put into a pool when we are shutting down.
-  ;; This can be used to build `borrow` functionality that will detect the
-  ;; case where we're trying to borrow during a shutdown, so we can return
-  ;; a sane error code.
-  [pool])
-
 (def supported-jruby-compile-modes
   #{:jit :force :off})
 
 (def SupportedJRubyCompileModes
```

Comment:
I don't think this or `supported-jruby-compile-modes` should be here... I think if necessary, the `CombinedJRubyPuppetConfig` could use this from `jruby-utils`, but my hope is we can avoid that by merging the configs together rather than enumerating all of the settings.